frontend/public/components/row-filters.tsx

```diff
@@ -0,0 +1,348 @@
+      const all = _.map(filter.items, 'id');
+      const recognized = _.intersection(newSelected, all);
+      (props.reduxIDs || []).forEach((id) =>
+        props.filterList(id, filter.type, { selected: new Set(recognized), all }),
+      );
+    });
+  };
+
+  const updateRowFilterSelected = (id) => {
+    const selectedNew = _.xor(selected, id);
+    updateRowFilters(selectedNew);
+    setQueryParameters(selectedNew);
+    setIsOpen(false);
+  };
+
+  const clearAllRowFilter = (f) => {
+    updateRowFilterSelected(_.intersection(filters[f], selected));
+  };
+
+  const onRowFilterSelect = (event) => {
+    event.preventDefault();
+    updateRowFilterSelected([event?.target?.id]);
+  };
+
+  return (
+    <>
+      <div className="co-list-filter-block">
+        {rowFilters.length > 0 && (
+          <Dropdown
+            dropdownItems={dropdownItems}
+            className="co-list-filter-block__row-dropdown"
+            onSelect={onRowFilterSelect}
+            isOpen={isOpen}
+            toggle={
+              <DropdownToggle onToggle={() => setIsOpen(!isOpen)} iconComponent={CaretDownIcon}>
+                <FilterIcon className="span--icon__right-margin" />
+                Filter
+              </DropdownToggle>
+            }
+          />
+        )}
+        <div className="co-list-fliter-block__user-input-filter">
+          {showLabelFilter && (
+            <DropdownInternal
+              items={FilterType}
+              onChange={(f) => setFilterType(FilterType[f])}
+              selectedKey={filterType}
+              title={filterType}
+            />
+          )}
+          {showNameFilter && (
+            <AutoCompleteInput
+              className="co-text-node co-m-label co-m-label--expand"
+              onSuggestionSelect={addLabelFilter}
+              showSuggestions={FilterType.LABEL === filterType}
+              // Text Input value
+              textValue={filterType === FilterType.NAME ? nameFilter : labelFilterText}
+              // Set text input value
+              setTextValue={updateSearchFilter}
+              placeholder={FilterType.NAME === filterType ? 'Search by name...' : 'app=frontend'}
+              data={data}
+            />
+          )}
+        </div>
+      </div>
+      <ChipGroup withToolbar defaultIsOpen={false} numChips={8}>
+        <ChipGroupToolbarItem key="name-category" categoryName={searchFilterValues.Name}>
+          {nameFilter !== '' && (
+            <Chip key="typehaed-chip" onClick={clearNameFilter}>
+              {nameFilter}
+            </Chip>
+          )}
+          {nameFilter !== '' && (
+            <span>
+              <Button variant="plain" aria-label="Close" onClick={clearNameFilter}>
+                <CloseIcon />
+              </Button>
+            </span>
+          )}
+        </ChipGroupToolbarItem>
+        {Object.keys(filters).map((key) => {
+          const selected_ = _.intersection(selected, filters[key]);
```

Comment:
Can we find a better name than `selected`? `selected` vs `selected_` gets confusing